src/IOHprofiler_csv_logger.cpp

```diff
@@ -0,0 +1,197 @@
+  }
+
+  if(interval_status()){
+    std::string idat_name = sub_directory_name + IOHprofiler_path_separator 
+                          + "IOHprofiler_f" + std::to_string(problem_id) + "_DIM"
+                          + std::to_string(dimension) + "_i1.idat";
+
+    if(idat.is_open()) idat.close();
+    idat.open(idat_name.c_str());
+    idat << dat_header << std::endl;
+  }
+
+  if(update_status()) {
+    std::string dat_name = sub_directory_name + IOHprofiler_path_separator 
+                            + "IOHprofiler_f" + std::to_string(problem_id) + "_DIM"
+                            + std::to_string(dimension) + "_i1.dat";
+
+    if(dat.is_open()) dat.close();
+    dat.open(dat_name.c_str());
+    dat << dat_header << std::endl;
+  }
+  
+};
+
+void IOHprofiler_csv_logger::write_line(size_t evaluations, double y, double best_so_far_y,
+                           double transformed_y, double best_so_far_transformed_y,
+                           std::vector<double> parameters) {
+  std::string written_line = std::to_string(evaluations) + " " + std::to_string(y) + " "
+                           + std::to_string(best_so_far_y) + " " + std::to_string(transformed_y) + " "
+                           + std::to_string(best_so_far_transformed_y);
+  for (std::vector<double>::iterator iter = parameters.begin(); iter != parameters.end(); ++iter) {
+    written_line = written_line + " " + std::to_string(*iter);
+  }
+  written_line += "\n";
+
+  if(complete_trigger()) {
+    if(!cdat.is_open()) IOH_error("*.cdat file is not open");
+    cdat << written_line;
+  }
+  if(interval_trigger(evaluations)) {
+    if(!idat.is_open()) IOH_error("*.idat file is not open");
+    idat << written_line;
+  }
+  if(update_trigger(transformed_y)) {
+    if(!dat.is_open()) IOH_error("*.dat file is not open");
+    dat << written_line;
+  }
+};
+
+void IOHprofiler_csv_logger::write_line(size_t evaluations, double y, double best_so_far_y,
+                           double transformed_y, double best_so_far_transformed_y) {
+  std::string written_line = std::to_string(evaluations) + " " + std::to_string(y) + " "
+                           + std::to_string(best_so_far_y) + " " + std::to_string(transformed_y) + " "
+                           + std::to_string(best_so_far_transformed_y) + "\n";
+
+  if(complete_trigger()) {
+    if(!cdat.is_open()) IOH_error("*.cdat file is not open");
+    cdat << written_line << std::endl;
+  }
+  if(interval_trigger(evaluations)) {
+    if(!idat.is_open()) IOH_error("*.idat file is not open");
+    idat << written_line;
+  }
+  if(update_trigger(transformed_y)) {
+    if(!dat.is_open()) IOH_error("*.dat file is not open");
+    dat << written_line;
+  }
+};
+
+
+/* The strategy for writing into needs to be discussed.
+ * Here we suppose that problems are to be tested by order of problem_id,
+ * which means that an infoFile will not be opened twice.*/
+void IOHprofiler_csv_logger::openInfo(int problem_id, int dimension) {
```

Comment:
This function is never called currently, so no .info file is created when running the test